### PR TITLE
Add Kustomize

### PR DESCRIPTION
**Pull request by levsha**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by levsha** [comment]
@conda-forge-admin, please ping conda-forge/staged-recipes

**Comment by levsha** [comment]
@conda-forge-admin, please ping conda-forge/help-go

File: recipes/kustomize/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+set -xeuo pipefail
+
+cd kustomize
+go build -ldflags \
+    "-X sigs.k8s.io/kustomize/api/provenance.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+    -X sigs.k8s.io/kustomize/api/provenance.version=v${PKG_VERSION}" \
+    .
```
Comment (xhochy) [inline recipes/kustomize/build.sh:9]:
```suggestion
    .

go-licenses save . --save_path="./license-files/"
```

File: recipes/kustomize/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  url: https://github.com/kubernetes-sigs/{{ name }}/archive/refs/tags/{{ name }}/v{{ version }}.tar.gz
+  sha256: 3c2c24971b29d96aac43853c090a5f042789d169bd41cbd9738c7979c5f09deb
+
+build:
+  skip: true  # [win]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('go') }}
```
Comment (xhochy) [inline recipes/kustomize/meta.yaml:18]:
```suggestion
    - {{ compiler('go') }}
    -  go-licenses
```

File: recipes/kustomize/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+about:
+  home: https://kustomize.io/
+  summary: 'Kubernetes native configuration management'
+  description: |
+    kustomize lets you customize raw, template-free YAML files for multiple purposes, leaving the original YAML untouched and usable as is.
+    kustomize targets kubernetes; it understands and can patch kubernetes style API objects. It's like make, in that what it does is declared in a file, and it's like sed, in that it emits edited text.
+  license: Apache-2.0
+  license_family: Apache
+  license_file: LICENSE
```
Comment (xhochy) [inline recipes/kustomize/meta.yaml:32]:
```suggestion
  license_file:
    - LICENSE
    - license-files/
```

**Review by xhochy — CHANGES_REQUESTED** [review]
This needs to bundle third-party licenses.

**Review by xhochy — APPROVED** [review]

**Pull request merged**